django/db/backends/postgresql/schema.py

```diff
@@ -297,7 +297,7 @@ def _alter_field(
         )
         # Added an index? Create any PostgreSQL-specific indexes.
         if (not (old_field.db_index or old_field.unique) and new_field.db_index) or (
-            not old_field.unique and new_field.unique
+            not old_field.unique and new_field._unique
```

Comment:
I don't believe that is a proper solution, you need to consider all cases and write it down to a correct logic formula. We have to consider here two cases:
1. when `_like` indexes should be added
2. when `_like` indexes should be re-created because they were removed in `super()._alter_field()`

Reply:
IMO, this is a regression in 9356f63a99957f01c14a9788617428a172a29fcb when the logic was simplified missing some cases.

Reply:
OK, thanks - I'll work on adding the missing tests.

Reply:
@felixxm I've added tests and updated the fix.

Reply:
Another option is what was mentioned in the issue: we could just use "IF NOT EXISTS" and not try to determine whether the _like index is already there or not.

Reply:
> @felixxm I've added tests and updated the fix.

We should not use `_unique` (as I said previously). Also, most of added tests seems to be redundant with the existing tests, or are redundant with each other e.g. `test_slugfields_change_primary_key` and `test_slugfields_change_primary_key_operations`.

Reply:
@felixxm What do you think of just using "IF NOT EXISTS" for creating the index? Then we wouldn't need to worry as much about all the different states of the indexes.